library/src/main/cpp/decoder_headers.h

```diff
@@ -50,4 +50,15 @@ ftyp_image_type get_ftyp_image_type(const uint8_t* data, uint32_t size) {
   return ftyp_image_type_no;
 }
 
+static uint8_t JXL_HEADER[] = "\0\0\0\x0cJXL \x0d\x0a......ftypjxl ";
```

Comment:
I haven't looked into the format specification, but shouldn't `\0\0\0\x0cJXL` be enough?

Reply:
~~It's probably sufficient~~
[found it!](https://github.com/libjxl/libjxl/blob/756a661541e08545c92495f166af56783ba9f167/lib/jxl/decode.cc#L127)